prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -110,14 +113,24 @@ trait Vectors extends Scalars with TimeUnits with Base {
       } else (metricName, None)
     }
 
+    private def parseBucketValue(value: String): Option[Double] =
+      if (value == "+Inf") Some(Double.PositiveInfinity) else Try(value.toDouble).toOption
+
     /**
      * Converts LabelMatches to ColumnFilters.  Along the way:
      * - extracts ::col column name expressions in metric names to columns
      * - removes ::col in __name__ label matches as needed
+     * Also extracts special _bucket_ histogram bucket filter
      */
     protected def labelMatchesToFilters(labels: Seq[LabelMatch]) = {
       var column: Option[String] = None
-      val filters = labels.map { labelMatch =>
+      var bucketOpt: Option[Double] = None
+      val filters = labels.filter { labelMatch =>
+        if (labelMatch.label == BucketFilterLabel) {
+          bucketOpt = parseBucketValue(labelMatch.value)
+          false
+        } else true
```

Comment:
Assuming `_type_` is going to be handled here as well ?

Reply:
It is handled in materialization of LPs.  Can address and improve that separately.